index.bs

```diff
@@ -795,6 +741,528 @@ the following list:
+(0). It has the same semantics as {{WebSocket}}'s [=WebSocket/ready state=], but is not exposed to
+JavaScript.
+
+
+<dl class="domintro non-normative">
+ : <code>|socket| = new {{WebSocketStream/constructor(url, options)|WebSocketStream}}(|url| [, |options| ]</code>
+ :: Creates a new {{WebSocketStream}} object, immediately establishing the associating WebSocket
+  connection.
+
+  |url| is a string giving the <a dfn spec=url>URL</a> over which the connection is established.
+  Only "`ws`" or "`wss`" schemes are allowed; others will cause a "{{SyntaxError}}"
+  {{DOMException}}. URLs with [=fragments=] will also cause such an exception.
+
+  The |options| argument is an object whose properties can be set as follows:
+
+  : {{WebSocketStreamOptions/protocols}}
+  :: An array of strings. If it is omitted, it is equivalent to an empty array. Each string in the
+   array is a subprotocol name. The connection will only be established if the server reports that
+   it has selected one of these subprotocols. The subprotocol names have to match the requirements
+   for elements that comprise the value of \`<a http-header>`Sec-WebSocket-Protocol`</a>\` fields as
+   defined by The WebSocket protocol.  [[!WSP]]
+
+  : {{WebSocketStreamOptions/signal}}
+  :: An {{AbortSignal}} that can be used to abort the handshake. After the handshake is complete the
+   signal does nothing.
+
+ : <code>|socket| . {{WebSocketStream/url}}</code>
+ :: Returns the [=WebSocketStream/url|URL=] that was used</a> to establish the WebSocket connection.
+
+ : <code>|socket| . {{WebSocketStream/opened}}</code>
+ :: Returns a {{promise}} which resolves when the handshake successfully completes, or rejects if
+  the handshake fails. On success, it resolves to an object with the following properties:
+
+  : {{WebSocketOpenInfo/readable}}
+  :: A {{ReadableStream}} that can be used to read messages from the server. Each chunk read
+   corresponds to one message. Text messages will be read as strings; binary messages will be read
+   as ArrayBuffer objects.
+
+   The stream can be closed by calling {{ReadableStream/cancel()}} on
+   {{WebSocketOpenInfo/readable}}. If the reason argument passed to {{ReadableStream/cancel()}} is a
+   {{WebSocketError}} object then {{WebSocketError/closeCode}} will be used as [=the WebSocket
+   connection close code=] and {{WebSocketError/reason}} will be used as [=the WebSocket connection
+   close reason=].
+
+   If no messages are read, or if messages are read slower than they are sent, then backpressure
+   will be applied and eventually the server will stop sending new messages.
+
+  : {{WebSocketOpenInfo/writable}}
+  :: A {{WritableStream}} that can be used to send messages to the server. Each chunk written will
+   be converted to one message. Strings will be sent as text messages; {{BufferSource}} chunks will
+   be sent as binary messages. Backpressure due to the network or server being behind will
```

Comment:
```suggestion
   be sent as binary messages. Backpressure due to the network or server being unable to process sufficient data will
```
Just felt like "being behind" is a bit colloquial...

Reply:
I went with "unable to process data fast enough".